src/server/getters/get-account-transaction-history.ts

```diff
@@ -0,0 +1,683 @@
+import * as t from "io-ts";
+import * as Knex from "knex";
+import { BigNumber } from "bignumber.js";
+import { Action, Coin, SortLimitParams, AccountTransactionHistoryRow } from "../../types";
+import { queryModifier } from "./database";
+
+export const GetAccountTransactionHistoryParams = t.intersection([
+  SortLimitParams,
+  t.type({
+    universe: t.string,
+    account: t.string,
+    earliestTransactionTime: t.number,
+    latestTransactionTime: t.number,
+    coin: t.string,
+    action: t.union([t.string, t.null, t.undefined]),
+  }),
+]);
+type GetAccountTransactionHistoryParamsType = t.TypeOf<typeof GetAccountTransactionHistoryParams>;
+
+function transformQueryResults(queryResults: Array<AccountTransactionHistoryRow<BigNumber>>) {
+  return queryResults.map((queryResult: AccountTransactionHistoryRow<BigNumber>) => {
+    if (queryResult.action === "BUY") {
+      queryResult.fee = queryResult.reporterFees.plus(queryResult.marketCreatorFees);
+      queryResult.total = new BigNumber(queryResult.quantity).times((queryResult.maxPrice).minus(queryResult.price));
+    } else if (queryResult.action === "SELL") {
+      queryResult.fee = queryResult.reporterFees.plus(queryResult.marketCreatorFees);
+      queryResult.total = new BigNumber(queryResult.quantity).times(queryResult.price);
+    } else if (queryResult.action === "DISPUTE" || queryResult.action === "INITIAL_REPORT") {
+      const divisor = new BigNumber(100000000000000000);
+      queryResult.quantity = new BigNumber(queryResult.quantity).dividedBy(divisor);
+    } else if (
+      queryResult.action === Action.CLAIM_MARKET_CREATOR_FEES || 
+      queryResult.action === Action.CLAIM_PARTICIPATION_TOKENS || 
+      queryResult.action === Action.CLAIM_TRADING_PROCEEDS || 
+      queryResult.action === Action.CLAIM_WINNING_CROWDSOURCERS
+    ) {
+      const divisor = new BigNumber(100000000000000000);
+      if (queryResult.action === Action.CLAIM_TRADING_PROCEEDS) {
+        let payoutAmount;
```

Comment:
You can simplify this a lot with the typesafety intact by doing the same thing we do here:
https://github.com/AugurProject/augur-node/blob/master/src/server/getters/get-winning-balance.ts#L46-L47